BookingsApi/BookingsApi.IntegrationTests/Database/Commands/EditJusticeUserCommandTests.cs

```diff
@@ -0,0 +1,122 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using AcceptanceTests.Common.Model.UserRole;
+using BookingsApi.DAL;
+using BookingsApi.DAL.Commands;
+using BookingsApi.DAL.Exceptions;
+using BookingsApi.Domain;
+using BookingsApi.Domain.Enumerations;
+using FluentAssertions;
+using Microsoft.EntityFrameworkCore;
+using NUnit.Framework;
+
+namespace BookingsApi.IntegrationTests.Database.Commands;
+
+public class EditJusticeUserCommandTests : DatabaseTestsBase
+{
+    private EditJusticeUserCommandHandler _commandHandler;
+
+    [SetUp]
+    public void Setup()
+    {
+        var context = new BookingsDbContext(BookingsDbContextOptions);
+        _commandHandler = new EditJusticeUserCommandHandler(context);
+    }
+
+    [TestCase(UserRoleId.Vho)]
+    [TestCase(UserRoleId.VhTeamLead)]
+    public async Task should_edit_a_justice_user(UserRoleId roleId)
+    {
+        // Arrange
+        var hearing = await SeedHearing();
+        var userToEdit = await SeedJusticeUser(hearing.Id, "testuser@hmcts.net");
+        
+        var command = new EditJusticeUserCommand(userToEdit.Id, userToEdit.Username, (int)roleId);
+        
+        // Act
+        await _commandHandler.Handle(command);
+        await using var db = new BookingsDbContext(BookingsDbContextOptions);
+        var justiceUser = db.JusticeUsers.FirstOrDefault(ju => ju.Username == command.Username);
+        
+        // Assert
+        justiceUser.Should().NotBeNull();
+        justiceUser.UserRoleId.Should().Be(command.RoleId);
+    }
+
+    [Test]
+    public async Task should_throw_an_exception_when_adding_a_justice_user_with_an_existing_username()
+    {
+        // Arrange
+        var id = Guid.NewGuid();
+        var command = new EditJusticeUserCommand(id, "testuser2@hmcts.net", (int)UserRole.VideoHearingsOfficer);
+
+        // Act & Assert
+        Assert.ThrowsAsync<JusticeUserNotFoundException>(async () =>
+        {
+            await _commandHandler.Handle(command);
+        }).Message.Should().Be($"Justice user with id {id} not found");
+    }
+    
+
+    private async Task<VideoHearing> SeedHearing() => await Hooks.SeedVideoHearing();
+
+    private async Task<JusticeUser> SeedJusticeUser(Guid allocatedHearingId, string username)
```

Comment:
I don't think you needed to recreate seeding a justice user, there's a aready a TestDataManager class which has that. Then the testbase you're inheriting handles cleaning it up after the test executs